docker/image_types.yaml

```diff
@@ -0,0 +1,67 @@
+# Defines the different types of docker images that FuzzBench uses and their
+# dependency graph.
+
+'base-image':
+  tag: 'base-image'
+  context: 'docker/base-image'
+
+'base-builder':
+  tag: 'base-builder'
+  context: 'docker/base-builder'
+  depends_on:
+    - 'base-image'
+
+'base-runner':
+  tag: 'base-runner'
+  context: 'docker/base-runner'
+  depends_on:
+    - 'base-image'
+
+'coverage-builder':
+  tag: 'builders/coverage'
+  context: 'fuzzers/coverage'
+  dockerfile: 'fuzzers/coverage/builder.Dockerfile'
+  depends_on:
+    - 'base-builder'
+
+'coverage-{benchmark}-builder':
```

Comment:
Does this need to be seperate from fuzzer-benchmark-builder?

Reply:
It doesn't need to be, but it makes things cleaner and simpler:
- otherwise, when we call `get_images_to_build(fuzzers, benchmark)`, then `fuzzers` would need to include `coverage` -- while it's not a fuzzer
- if we considered "coverage" as a fuzzer, `get_images_to_build` would also generate a coverage-benchmark-runner "image prototype" due to the '{fuzzer}-{benchmark}-runner' type -- and we don't have / need such image